sdk/typespec/typespec_macros/README.md

```diff
@@ -0,0 +1,82 @@
+# TypeSpec Macros crate for Rust
+
+The TypeSpec Macros crate provides procedural macros for [TypeSpec](https://typespec.io)-generated client libraries. These macros simplify the implementation of common patterns required when working with TypeSpec-generated code.
+
+[Source code] | [Package (crates.io)] | [API reference documentation] | [TypeSpec documentation]
+
+## Getting started
+
+> **Note:** This crate should not be used directly. Users should depend on the `typespec_client_core` crate instead.
+
+### Install the package
+
+If you need to use this crate directly, install the TypeSpec Macros crate for Rust with cargo:
+
+```bash
+cargo add typespec_macros
+```
+
+## Key concepts
+
+This crate provides the following derive macros:
+
+-   `Model`: A derive macro that implements the `Model` trait, allowing a type to be deserialized from an HTTP response body.
+-   `SafeDebug`: A derive macro that implements debug formatting in a way that avoids leaking personally identifiable information (PII).
+
+### The Model derive macro
+
+The `Model` derive macro is used to implement the `Model` trait for structs that represent data returned from an API. It handles the deserialization of HTTP response bodies into your model types.
+
+### The SafeDebug derive macro
+
+The `SafeDebug` derive macro creates a `Debug` implementation that respects the `#[sensitive]` attribute on struct fields. Fields marked with this attribute will not have their values printed in debug output, protecting potentially sensitive information.
+
+## Examples
+
+### Using the Model derive macro
+
+```rust
+use typespec_macros::Model;
+use serde::Deserialize;
+
+#[derive(Model, Deserialize)]
+pub struct User {
+    pub id: String,
+    pub name: String,
+    pub email: String,
+}
+```
+
+### Using the SafeDebug derive macro
+
+```rust
+use typespec_macros::SafeDebug;
+
+#[derive(SafeDebug)]
+pub struct Credentials {
+    pub username: String,
+    #[sensitive]
```

Comment:
This isn't the correct syntax for `#[derive(SafeDebug)]`.

By default `SafeDebug` redacts all entries and you need to mark elements as `#[safe(true)]` to have them included in the output.